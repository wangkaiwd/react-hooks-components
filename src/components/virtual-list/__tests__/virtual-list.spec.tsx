import { fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import VirtualList from "../index";

const data: any[] = [];
for (let i = 0; i < 100; i++) {
  data.push({ id: i });
}
describe("VirtualList:", () => {
  const genList = () => render(
    <VirtualList height={300} itemHeight={30} data={data} itemKey={"id"}>
      {(item, i) => <div style={{ height: "30px" }}>{i}</div>}
    </VirtualList>
  );
  it("should render successfully", () => {
    const tree = renderer.create(
      <VirtualList height={300} itemHeight={30} data={data} itemKey={"id"}>
        {(item, i) => <div style={{ height: "30px" }}>{i}</div>}
      </VirtualList>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should only render items in visible view", () => {
    const { container } = genList();
    const wrapper = container.querySelector(".ant-virtual-list-wrapper");
    expect(wrapper?.children.length).toBeLessThan(data.length);
  });
  it("should only render items in visible view when list container scroll", () => {
    const { container } = genList();
    fireEvent.scroll(container.firstChild!, { target: { scrollTop: 10 } });
    const wrapper = container.querySelector(".ant-virtual-list-wrapper");
    expect(wrapper?.children.length).toBeLessThan(data.length);
  });
});
