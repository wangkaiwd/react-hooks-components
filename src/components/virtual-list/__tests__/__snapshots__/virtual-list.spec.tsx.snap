// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualList: should render successfully 1`] = `
<div
  className="ant-virtual-list"
  onScroll={[Function]}
  style={
    Object {
      "height": "300px",
    }
  }
>
  <div
    className="ant-virtual-list-inner"
    style={
      Object {
        "height": "3000px",
      }
    }
  >
    <div
      className="ant-virtual-list-wrapper"
      style={
        Object {
          "transform": "translateY(0px)",
        }
      }
    >
      <div
        style={
          Object {
            "height": "30px",
          }
        }
      >
        0
      </div>
      <div
        style={
          Object {
            "height": "30px",
          }
        }
      >
        1
      </div>
      <div
        style={
          Object {
            "height": "30px",
          }
        }
      >
        2
      </div>
      <div
        style={
          Object {
            "height": "30px",
          }
        }
      >
        3
      </div>
      <div
        style={
          Object {
            "height": "30px",
          }
        }
      >
        4
      </div>
      <div
        style={
          Object {
            "height": "30px",
          }
        }
      >
        5
      </div>
      <div
        style={
          Object {
            "height": "30px",
          }
        }
      >
        6
      </div>
      <div
        style={
          Object {
            "height": "30px",
          }
        }
      >
        7
      </div>
      <div
        style={
          Object {
            "height": "30px",
          }
        }
      >
        8
      </div>
      <div
        style={
          Object {
            "height": "30px",
          }
        }
      >
        9
      </div>
      <div
        style={
          Object {
            "height": "30px",
          }
        }
      >
        10
      </div>
    </div>
  </div>
</div>
`;
